### PR TITLE
Use base64url encoding for oauth2 state parameter

### PR DESCRIPTION
**Pull request by zhaohuabing**
Related envoy pr: https://github.com/envoyproxy/envoy/pull/37473

### CODE REVIEW

File: single-page-app/verify.sh [hunk 1]
```diff
@@ -217,12 +221,12 @@ test_auth () {
...
 }
 
-get_js () {
-    _curl -k "https://localhost:${PORT_PROXY}" \
-        | grep "assets/index" \
-        | grep -oP '<script type="module" crossorigin src="/assets/[^"]+"></script>' \
-        | grep -oP '/assets/[^"]+' \
-        | sed 's/\/assets\///;s/".*//'
+get_js() {
+    _curl -k "https://localhost:${PORT_PROXY}" |
```
Comment (phlax) [inline single-page-app/verify.sh:225]: i dont overly like most of the format changes but in this case i **strongly** disagree - mho is that code conjunctions _should_ be at start of line - i think it makes the code an order of magnitude easier to read

**Review by phlax — COMMENTED** [review]
can we seperate (or remove) the formatting changes - it would make the changes easier to review

**Comment by zhaohuabing** [comment]
@phlax pologies for the unintended shell format changes. They were inadvertently introduced by the formatter in my VS Code. I've now reverted the changes to restore the original format.

**Comment by zhaohuabing** [comment]
SkyWalking failed, which is irrelevant to this PR. How can I rerun the test?

**Review by phlax — APPROVED** [review]
lgtm, thanks @zhaohuabing

**Pull request merged**